File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03558/p03558_num4_parsed.py
def find_smallest_sum(K):
    # Find the smallest possible multiple of K
    smallest_multiple = 0
    while smallest_multiple == 0:
        i = 1
        while True:
            if (K * i) % 10 == 0 or (K * i) % 10 == 1:
                smallest_multiple = K * i
                break
            i += 1

    # Calculate the sum of the digits
    sum_of_digits = 0
    for digit in str(smallest_multiple):
        sum_of_digits += int(digit)

    return sum_of_digits


# Get the input
K = int(input())

# Call the function and print the result
print(find_smallest_sum(K))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03558/p03558_num4_parsed.cpp
#include <iostream>
#include <vector>

int getSmallestSum(int K) {
    std::vector<int> remainders(K, 0); // vector to keep track of remainders
    std::vector<int> digits(K, -1); // vector to store the digit at each multiple
    digits[0] = 0; // the first multiple is a single digit

    int num = 0; // current multiple
    int sum = 0; // sum of the digits

    // find the smallest multiple of K with each possible digit
    for (int i = 1; i < K; i++) {
        num = (num * 10 + 1) % K; // new multiple

        if (digits[num] != -1) {
            // found a cycle, the smallest multiple with each digit has been found
            // break the cycle and calculate the smallest sum
            for (int j = digits[num]; j < i; j++) {
                sum += digits[j];
            }
            break;
        } else {
            digits[num] = i;
        }
    }

    return sum;
}

int main() {
    int K;
    std::cin >> K;

    int smallestSum = getSmallestSum(K);
    std::cout << smallestSum;

    return 0;
}